providers/os/registry/registryhandler_windows.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build windows
+// +build windows
+
+package registry
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+var (
+	advapi32 = syscall.NewLazyDLL("advapi32.dll")
+	// note: we're using the W (RegLoadKeyW and NOT RegLoadKeyA) versions of these functions to work with UTF16 strings
+	regLoadKey   = advapi32.NewProc("RegLoadKeyW")
+	regUnloadKey = advapi32.NewProc("RegUnLoadKeyW")
+)
+
+func LoadRegistrySubkey(key, path string) error {
+	keyPtr, err := syscall.UTF16PtrFromString(key)
+	if err != nil {
+		return err
+	}
+	pathPtr, err := syscall.UTF16PtrFromString(path)
+	if err != nil {
+		return err
+	}
+	ret, _, err := regLoadKey.Call(syscall.HKEY_LOCAL_MACHINE, uintptr(unsafe.Pointer(keyPtr)), uintptr(unsafe.Pointer(pathPtr)))
+	// the Microsoft docs indicate that the return value is 0 on success
```

Comment:
this is funny, what is the `err` for then? 😂

Reply:
also, what if ret is `0` but `err` is nil?

Would the caller of `LoadRegistrySubkey()` be confused if it was a success of failure?

Reply:
The err is to be thrown when the `ret` is not 0. There's always an err, even on success, so we cannot rely on the usual `err != nil` check:
https://go.dev/src/syscall/dll_windows.go (see 151)